### ISSUE TITLE
Add Crossref.org as source for web search

### ISSUE THREAD

**Issue by matthiasgeiger** [JabRef/jabref#2455]
As [complained here at the forum](http://discourse.jabref.org/t/the-google-scholar-fetcher-should-also-fetch-the-doi/360/6?u=matthiasgeiger) the Google Scholar BibTeX quality is not the best. As there are no "good" databases for all research disciplines present in JabRef we should add the crossref.org service to the web search fetchers (we are already using it for DOItoBibTeX).

Implementation should be straightforward but I've currently no time to do it...

REST API Documentation: https://github.com/CrossRef/rest-api-doc/blob/master/rest_api.md
Web search: http://search.crossref.org/

**Comment by stefan-kolb** [comment]
Wishes for fulltext fetcher will be collected at #2581.

**Comment by wujastyk** [comment]
Tried out the new crossref fetcher in 4.0.0 beta today, but it gave an error message:

![screenshot from 2017-05-02 11-04-31](https://cloud.githubusercontent.com/assets/762246/25629568/3264e97c-2f27-11e7-94cc-66f8cc58ddd3.png)

Other fetchers, GVK etc., are working, and the network is fine.

**Comment by matthiasgeiger** [comment]
Unfortunately the crossref service is quite unstable 😥 there are quite often some problems at their Server side which prevent correct fetching of entries...

**Comment by wujastyk** [comment]
That's annoying, of course.  But I don't think the instability of Crossref accounts for the problem we're seeing, though, because some searches (e.g., "zysk") _always_ give a result, and others ("wujastyk") _always_ give an error.  Plus, searching for a problem-term at crossref.org itself works fine.

**Comment by tobiasdiez** [comment]
@wujastyk thanks for the report! There was a problem in our parsing logic which should be resolved now. 
Could you please check the build from http://builds.jabref.org/master/. (be advised that some parts are still a bit unstable, especially the groups panel and the entry editor)

**Comment by wujastyk** [comment]
Yes, that fixed the problem!  Many thanks!

When I ran JabRef-4.0.0-dev--snapshot--2017-05-03--master--a16e6123d.jar, I
had to install the Oracle 8 version of java.  It wouldn't run with
java-1.8.0-openjdk-amd64.  Error message:

$ java -jar JabRef-4.0.0-dev--snapshot--2017-05-03--master--a16e6123d.jar
Error: Could not find or load main class org.jabref.JabRefMain

With Oracle 8, the program ran, but I got these messages, which didn't stop
the program working just fine:

java -jar JabRef-4.0.0-dev--snapshot--2017-05-03--master--a16e6123d.jar
AI: INFO 03-05-2017 15:52, 20: Configuration file has been successfully
found as resource
15:52:39.976 [AWT-EventQueue-0] INFO
 org.jabref.migrations.PreferencesMigrations - Migrating old preferences.
15:52:45.633 [AWT-EventQueue-0] INFO
 org.jabref.logic.importer.OpenDatabase - Opening:
/home/dom/Dropbox/localtexmf/bibtex/bib/biblio4-utf8.bib
File: grouptree.fxml not found, attempting with camel case
File: grouptree.css not found, attempting with camel case
15:52:48.792 [AWT-EventQueue-0] WARN  org.jabref.model.entry.BibEntry -
Could not parse date 1982/
ANTLR Tool version 4.5.3 used for code generation does not match the
current runtime version 4.7ANTLR Runtime version 4.5.3 used for parser
compilation does not match the current runtime version 4.7ANTLR Tool
version 4.5.3 used for code generation does not match the current runtime
version 4.7ANTLR Runtime version 4.5.3 used for parser compilation does not
match the current runtime version 4.7ANTLR Tool version 4.5.3 used for code
generation does not match the current runtime version 4.7ANTLR Runtime
version 4.5.3 used for parser compilation does not match the current
runtime version 4.7ANTLR Tool version 4.5.3 used for code generation does
not match the current runtime version 4.7ANTLR Runtime version 4.5.3 used
for parser compilation does not match the current runtime version 4.7


​
​Best wishes, and thanks again,
Dominik​
